cc-statistics/README.md

```diff
@@ -0,0 +1,228 @@
+
+You are now ready to run the notebook! Simply load it and take a look. Double check any variables with pathing.
+
+
+### Notebook
+
+For any existing code, remember to double check any variables related to pathing. Change them to adhere to your own directory's structure, if required.
+
+The current existing notebook code will process and plot an indegree frequency scatter plot, using the `.indegree` file which for each line i, the i'th line has contains the number of Pay level Domains with i indegrees. 
+
+If you would like to analyze the metrics of another graph characteristic (such as outdegrees), it is as simple as changing the `file_name` variable so that it points to the correct file.
+
+
+# pyWebGraph
+
+### Introduction
+Another interesting tool that you may look into is [pyWebGraph](https://github.com/mapio/py-web-graph), a program developed by [Massimo Santini](https://santini.di.unimi.it/) which interfaces Jython with WebGraph. This tool makes the exploration of WebGraphs very easy. The following sections will outline how to get started using pyWebGraph.
+
+### Requirements
+
+Python (2.x)
+
+### Quick Start
+
+
+#### Rebuilding String maps
+In order to get pyWebGraph working we need to (rebuild the string maps)[http://law.di.unimi.it/tutorial.php].
+Following the above steps to get the notebook started, you should have a vertices file (cc-main-2020-feb-mar-may-domain-vertices-domain.txt.gz) which contains a mapping from a node id to a reversed domain name. We however need to attain a mapping from reversed domain name to numbers. To achieve this we will use the (minimal perfect hashing algorithm)[http://sux4j.di.unimi.it/docs/it/unimi/dsi/sux4j/mph/package-summary.html]. 
+
+Running the following command will produce a `mph` (minimal perfect hash) file that contains mapping from reversed domain name to number.
+
+```
+zcat cc-main-2020-feb-mar-may-domain-vertices.txt.gz \
+  | cut -f2 \
+  | $WG it.unimi.dsi.sux4j.mph.GOV4Function cc-main-2020-feb-mar-may-domain.mph
+``` 
+
+However, we would also like to have the functionality of determining if a string is actually present as a domain in the map. We will produce a `String Map` to do so.
+
+Running the following command will produce a `smph` file (string map perfect hash) that has the outlined functionality.
+
+```
+zcat cc-main-2020-feb-mar-may-domain-vertices.txt.gz \
+  | cut -f2 \
+  | $WG it.unimi.dsi.util.ShiftAddXorSignedStringMap cc-main-2020-feb-mar-may-domain.mph cc-main-2020-feb-mar-may-domain.smph
+```
+
+We will also need a front coded list to map node numbers to URLs:
```

Comment:
The front coded list isn't used here, for the domain graph it will also fail because it does not meet the sorting requirement (domain names are not lexically sorted because of the dot/hyphen issue).

Reply:
Note: we spoke and FCL is needed, but immutable external prefix map is not. Double check on call.

https://gist.github.com/sebastian-nagel/95a086d7e8c76adc81d647258926e281

Reply:
Yes. But I will also verify while testing everything.